### ISSUE TITLE
add an option to publish to jdbc database.

### ISSUE THREAD

**Issue by AliTajeldin** [TresAmigosSD/SMV#705]
While sqoop is great at parallel export of hdfs files to databases, it would be convenient for users to use an smv level `--publish-jdbc` or something similar.

**Note:** We would require a config that determines the flavor of the actual DB used as the DDL for SQL databases is not really standard.

**Comment by AliTajeldin** [comment]
Increasing priority as this seems to also be the only way to change hive credentials (See #704)

**Comment by laneb** [comment]
Branch i705_jdbc now has support for 

1. Reading from a JDBC connection with SmvJdbcTable
2. Exporting through a JDBC connection using --publish-jdbc

For both uses, I have delegated to Spark's `DataFrameWriter`. Based on discussion with @AliTajeldin, writing through JDBC uses sets `DataFrameWriter's` save mode to `SaveMode.Append`, which means that `DataFrames` should be inserted into tables instead of overwriting them. *However*, this behavior doesn't seem entirely consistent - I was successfully able to insert into existing `MySQL` tables over JDBC, but when I tried to do the same with Derby tables there was a failure that implied that JDBC was trying to create the table. As the use-case inspiring this feature is to write to Hive over JDBC, we should verify that the insertion works as expected for Hive.
